### ISSUE TITLE
NavigationView.SelectedItem doesn't update when Two-Way binding is set

### ISSUE THREAD

**Issue by SuperJMN** [unoplatform/uno#2538]
NavigationView.SelectedItem doesn't reflect the currently selected item.

## Current behavior

Given this code (XAML and ViewModel code)

```
<Page
    x:Class="App4.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:App4"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <Page.DataContext>
        <local:ViewModel />
    </Page.DataContext>

    <NavigationView MenuItemsSource="{Binding Items}" SelectedItem="{Binding SelectedItem, Mode=TwoWay}" />
</Page>

```
```
namespace App4
{
    public class ViewModel
    {
        public ViewModel()
        {
            Items = new List<string>()
            {
                "Section 1",
                "Section 2",
            };

            SelectedItem = Items.First();
        }

        public string SelectedItem { get; set; }

        public List<string> Items { get; set; }
    }
}
```

When the SelectedItem in the ViewModel changes, nothing happens in the NavigationView.

> The NavigationView has a list of menu items. In this list, nothing gets selected when `SelectedItem` is set to one of the existing menu items

## Expected behavior

Changing the SelectedItem to one of the existing items should make the corresponding menu item to be selected.

## How to reproduce it (as minimally and precisely as possible)

This is a minimal repro: [Repro.zip](https://github.com/unoplatform/uno/files/4112287/Repro.zip)

## Environment

Package Version(s):

2.1.0-dev.872

Affected platform(s):

- [ ] iOS
- [ ] Android
- [X] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [X] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [X] Resharper (version: )

**Comment by jeromelaban** [comment]
Will be fixed by #2462

**Comment by SuperJMN** [comment]
Sadly it still reproduces with v2.1.0-dev.941. You can repro by cloning this: https://github.com/SuperJMN/UnoPlatform.AppTemplate. Take into consideration that the Backend project should be running before the client app (multiple startup) and that the affected head is WASM.

**Comment by LHuss** [comment]
Seems to have been fixed by #4096
